### DISCUSSION TITLE
Briefly define Dijkstra's algorithm.

### DISCUSSION THREAD

**Discussion by ankit-akash** [ankit-akash/DSA#45] [Q&A] [answered]
brief

**Comment by fcasamento** [comment, marked as answer]
Given a graph and a source vertex in the graph, find the shortest paths from the source to all vertices in the given graph.

Dijkstra’s algorithm is very similar to Prim’s algorithm for minimum spanning tree. 

Like Prim’s MST, generate a SPT (shortest path tree) with a given source as a root. Maintain two sets, one set contains vertices included in the shortest-path tree, other set includes vertices not yet included in the shortest-path tree. At every step of the algorithm, find a vertex that is in the other set (set not yet included) and has a minimum distance from the source.

![220px-Dijkstra_Animation](https://user-images.githubusercontent.com/19937845/205523094-2aa2ed52-94da-41c0-bf90-b7292c8aba4c.gif)

**Reply by fcasamento** [reply]
@ankit-akash Mark this as an answer as well, we can help each other. 🙏🏻 Pls
